zeroex/orderwatch/order_watcher.go

```diff
@@ -1142,35 +1159,52 @@ func (w *Watcher) generateOrderEventsIfChanged(
+				if validationBlockTimestamp.Before(expiration) {
+					w.rewatchOrder(ordersColTxn, order, order.FillableTakerAssetAmount)
+					orderEvent := &zeroex.OrderEvent{
+						Timestamp:                validationBlockTimestamp,
+						OrderHash:                order.Hash,
+						SignedOrder:              order.SignedOrder,
+						FillableTakerAssetAmount: order.FillableTakerAssetAmount,
+						EndState:                 zeroex.ESOrderUnexpired,
+					}
+					orderEvents = append(orderEvents, orderEvent)
+				}
 			}
-			orderEvents = append(orderEvents, orderEvent)
-		} else if oldFillableAmount.Cmp(big.NewInt(0)) == 1 && !oldAmountIsMoreThenNewAmount {
-			// The order is now fillable for more then it was before. E.g.: A fill txn reverted (block-reorg)
-			// Update order in DB and emit event
-			order.FillableTakerAssetAmount = newFillableAmount
-			w.updateOrderDBEntry(ordersColTxn, order)
-			orderEvent := &zeroex.OrderEvent{
-				Timestamp:                validationBlockTimestamp,
-				OrderHash:                acceptedOrderInfo.OrderHash,
-				SignedOrder:              order.SignedOrder,
-				EndState:                 zeroex.ESOrderFillabilityIncreased,
-				FillableTakerAssetAmount: acceptedOrderInfo.FillableTakerAssetAmount,
-				ContractEvents:           orderHashToEvents[order.Hash],
+			if oldFillableAmount.Cmp(newFillableAmount) == 0 {
+				// No important state-change happened
+			} else if oldFillableAmount.Cmp(big.NewInt(0)) == 1 && oldAmountIsMoreThenNewAmount {
+				// Order was filled, emit event and update order in DB
+				order.FillableTakerAssetAmount = newFillableAmount
+				w.updateOrderDBEntry(ordersColTxn, order)
+				orderEvent := &zeroex.OrderEvent{
+					Timestamp:                validationBlockTimestamp,
+					OrderHash:                acceptedOrderInfo.OrderHash,
+					SignedOrder:              order.SignedOrder,
+					EndState:                 zeroex.ESOrderFilled,
+					FillableTakerAssetAmount: acceptedOrderInfo.FillableTakerAssetAmount,
+					ContractEvents:           orderHashToEvents[order.Hash],
+				}
+				orderEvents = append(orderEvents, orderEvent)
+			} else if oldFillableAmount.Cmp(big.NewInt(0)) == 1 && !oldAmountIsMoreThenNewAmount {
+				// The order is now fillable for more then it was before. E.g.: A fill txn reverted (block-reorg)
+				// Update order in DB and emit event
+				order.FillableTakerAssetAmount = newFillableAmount
+				w.updateOrderDBEntry(ordersColTxn, order)
+				orderEvent := &zeroex.OrderEvent{
+					Timestamp:                validationBlockTimestamp,
+					OrderHash:                acceptedOrderInfo.OrderHash,
+					SignedOrder:              order.SignedOrder,
+					EndState:                 zeroex.ESOrderFillabilityIncreased,
+					FillableTakerAssetAmount: acceptedOrderInfo.FillableTakerAssetAmount,
+					ContractEvents:           orderHashToEvents[order.Hash],
+				}
+				orderEvents = append(orderEvents, orderEvent)
 			}
```

Comment:
Optional: There is some code duplication in these blocks that can be removed.

Reply:
Will leave as is so we don't need to add an additional conditional nesting.